crates/conformance-test-utils/src/hive/genesis.rs

```diff
@@ -23,6 +38,143 @@ impl HiveGenesisConfig {
+    });
+
+    // Set the Kakarot contracts address and proxy class hash
+    let kakarot_address = FieldElement::from_hex_be(KAKAROT_ADDRESSES).unwrap(); // Safe unwrap, 0x9001
+    let blockhash_registry_address = FieldElement::from_hex_be(BLOCKHASH_REGISTRY_ADDRESS).unwrap(); // Safe unwrap, 0x9002
+    let account_proxy_class_hash = *kakarot_contracts.get("proxy").expect("Failed to get proxy class hash");
+    let contract_account_class_hash =
+        *kakarot_contracts.get("contract_account").expect("Failed to get contract_account class hash");
+    let eoa_class_hash = *kakarot_contracts.get("externally_owned_account").expect("Failed to get eoa class hash");
+
+    // Add Kakarot contracts to Loader
+    madara_loader.contracts.push((
+        HexFelt(kakarot_address),
+        HexFelt(*kakarot_contracts.get("kakarot").expect("Failed to get kakarot class hash")),
+    ));
+    madara_loader.contracts.push((
+        HexFelt(blockhash_registry_address),
+        HexFelt(*kakarot_contracts.get("blockhash_registry").expect("Failed to get blockhash_registry class hash")),
+    ));
+
+    // Set storage keys of Kakarot contract
+    // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/constants.cairo
+    let storage_keys = [
+        ("native_token_address", FieldElement::from_hex_be(STARKNET_NATIVE_TOKEN).unwrap()),
+        ("contract_account_class_hash", contract_account_class_hash),
+        ("externally_owned_account", eoa_class_hash),
+        ("account_proxy_class_hash", account_proxy_class_hash),
+        ("blockhash_registry_address", blockhash_registry_address), // Safe unwrap 0x9002
+    ];
+
+    storage_keys.iter().for_each(|(key, value)| {
+        let storage_tuple = genesis_set_storage_starknet_contract(kakarot_address, key, &[], *value, 0);
+        madara_loader
+            .storage
+            .push(unsafe { std::mem::transmute::<((Felt, Felt), Felt), ((HexFelt, HexFelt), HexFelt)>(storage_tuple) });
+    });
+
+    // Add Hive accounts to loader
+    // Convert the EVM accounts to Starknet accounts using compute_starknet_address
+    // Sort by key to ensure deterministic order
+    let mut hive_accounts: Vec<(reth_primitives::H160, AccountInfo)> = hive_genesis.alloc.into_iter().collect();
+    hive_accounts.sort_by_key(|(address, _)| *address);
+    hive_accounts.iter().for_each(|(evm_address, account_info)| {
+        // Use the given Kakarot contract address and declared proxy class hash for compute_starknet_address
+        let starknet_address = compute_starknet_address(
+            kakarot_address,
+            account_proxy_class_hash,
+            FieldElement::from_byte_slice_be(evm_address.as_bytes()).unwrap(), /* safe unwrap since evm_address
+                                                                                * is 20 bytes */
+        );
+        // Push to contracts
+        madara_loader.contracts.push((HexFelt(starknet_address), HexFelt(account_proxy_class_hash)));
```

Comment:
I think we will have a problem here: you are setting the address of the account to be a proxy, but you still need to set [the proxy implementation](https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/proxy/upgradable.cairo#L11) (if it's an EOA to eoa class hash, if it's a contract i.e. contains bytecode, to the contract account class hash).
So you will need to modify the storage of the account based on what type of account it is.

Reply:
Gotcha I missed this, thanks will add this logic to setting the proxy implementation